lib/libutee/include/user_ta_header.h

```diff
@@ -32,7 +32,16 @@
 #define TA_FLAG_DEVICE_ENUM		(1 << 9)  /* without tee-supplicant */
 #define TA_FLAG_DEVICE_ENUM_SUPP	(1 << 10) /* with tee-supplicant */
 
-#define TA_FLAGS_MASK			GENMASK_32(10, 0)
+	/*
+	 * The flag is to determine if TA get panic or not when an unsupported
+	 * entry function ID is called. If this flag is set, it returns an error
+	 * TEE_ERROR_NOT_IMPLEMENTED instead of calling TEE_Panic().
+	 *
+	 * This is to ensure backward compatibility when adding a new entry
+	 * function to user TA entry.
+	 */
+#define TA_FLAG_ENTRY_NO_PANIC		BIT(11)  /* no panic in TA entry */
+#define TA_FLAGS_MASK			GENMASK_32(11, 0)
```

Comment:
This patch should also remove the `TEE_Panic(0);` in `__utee_entry()`, otherwise it is not safe to call it. And replace it by some code outside the TA (in `thread_enter_user_mode()` perhaps?) that would detect the error condition that the function ID was not recognized and panic (kill) the TA. This last part may be a bit complex, I'm not sure. Maybe a function ID could be allocated to ask the TA to call `TEE_Panic()`.

Reply:
Understand your concern that we have TA_FLAG_ENTRY_NO_PANIC in TA header, however it gets panic when a non-recognized coming.
Can we handle in this way?
in `__utee_entry() :: default`, we check `ta_head.flags & TA_FLAG_ENTRY_NO_PANIC`.
if it's true, return `TEE_ERROR_NOT_IMPLEMENTED`;
else `TEE_Panic(0)` to kill TA.

Reply:
The problem with what you are suggesting is that `utee_entry()` is in libutee which is typically linked statically in TAs. So older TAs would still panic.

Reply:
I'm confused :confused: I think the flag implemented here has a slightly different meaning that what I had in mind. I need to take a closer look.

Reply:
in the inline comment, maybe be more explicit:
> The flag is to determine if TA panics or not when an unsupported
> entry function ID is called. If this flag is set, it returns an error
> TEE_ERROR_NOT_IMPLEMENTED instead of **entering TA**
> **context and panicking as entry function ID is not handled**.